packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -40,6 +42,16 @@ export default class ConfigParser {
     ) {}
 
     autoCompile() {
+        /**
+         * Checks tsconfig.json path, throws error if it doesn't exist
+         */
+        if (this._config.autoCompileOpts?.tsNodeOpts?.project) {
+            const tsconfigPath = path.resolve(this._config.autoCompileOpts.tsNodeOpts.project)
+            if (!fs.existsSync(tsconfigPath)) {
```

Comment:
Let's not use `existsSync` as it blocks the main thread and better practice here is to use `await fs.exists(...)` from `fs/promise` and make the method async